source/common/router/router.cc

```diff
@@ -630,6 +605,57 @@ void Filter::onUpstreamReset(UpstreamResetType type,
   }
 }
 
+bool Filter::maybeRetryReset(Http::StreamResetReason reset_reason) {
+  // We don't retry on a global timeout or if we already started the response.
+  if (!downstream_response_started_ && retry_state_) {
```

Comment:
nit: now that you moved this out, I would do an early logic check and return, and then unindent the rest of the logic.